Add Knowledge Base docs (#13)
optakt/optakt.github.io: content/knowledge_base/consensus/avalanche/concepts.md

> # Consensus algorithm concepts behind Avalanche

Reviewer: ```suggestion
# Avalanche Consensus Algorithm Concepts
```

WDYT of changing it to this? For the capitalization of titles in our markdown documents, please follow the first point in the [Markdown style guidelines](https://github.com/optakt/optakt.github.io/blob/master/content/style/markdown.md#general).

Author: This has been fixed https://github.com/optakt/optakt.github.io/pull/13/commits/211354094b1da3445d32b93c04cf950b6a935394#diff-7b8cdfdde84c7352f51f3cc642d28c90f9d1283870319c81956812353289fb32R1-R2

> ## Snow Algorithm group of protocols

Reviewer: ```suggestion
## Snow Algorithm Group of Protocols
```

Author: Changed section names to Chicago style here and everywhere else
https://github.com/optakt/optakt.github.io/pull/13/commits/211354094b1da3445d32b93c04cf950b6a935394#diff-7b8cdfdde84c7352f51f3cc642d28c90f9d1283870319c81956812353289fb32R3-R4

> A group of protocols with a strong probabilistic safety guarantee in the presence or Byzantine nodes.

Reviewer: ```suggestion
A group of protocols with a strong probabilistic safety guarantee in the presence of Byzantine nodes.
```

Author: Fixed https://github.com/optakt/optakt.github.io/pull/13/commits/0a1cb902f3417c3bc20f9197df71d3d65a1b6421#diff-7b8cdfdde84c7352f51f3cc642d28c90f9d1283870319c81956812353289fb32R5-R6
Thanks!

### Description

> The core concept of operating is execution of repeated sampling the network at random, ultimately leading to correct nodes behavior to obtain a common statement (decision, transacation value, outcome). This mechanism effectively brings the system to an irreversible state - meaning that a large potion of the network accepted a decision or a statement. So any such conflicting to that statement would be accepted with a negligible probability ε or smaller.

Reviewer: ```suggestion
### Description

The core concept of operating is execution of repeated sampling the network at random, ultimately leading to correct nodes behavior to obtain a common statement (decision, transaction value, outcome). This mechanism effectively brings the system to an irreversible state — meaning that a large potion of the network accepted a decision or a statement. So any such conflicting to that statement would be accepted with a negligible probability ε or smaller.
```

See Markdown style guidelines:

> Keep extra newlines between paragraphs and sections to make the source easier to read.

Also when using an `em dash` for punctuation, please make sure to use the `—` character which is specifically made for that purpose

Author: This description has been updated https://github.com/optakt/optakt.github.io/pull/13/commits/0a1cb902f3417c3bc20f9197df71d3d65a1b6421#diff-7b8cdfdde84c7352f51f3cc642d28c90f9d1283870319c81956812353289fb32R7-R13
But I fixed rest of the places

TODO: finish
TODO: add references

> ### Compare with Nakamoto type of consensus:

Reviewer: ```suggestion
### Comparison with Nakamoto Consensus:
```

Author: Fixed, thanks!
https://github.com/optakt/optakt.github.io/pull/13/commits/0a1cb902f3417c3bc20f9197df71d3d65a1b6421#diff-7b8cdfdde84c7352f51f3cc642d28c90f9d1283870319c81956812353289fb32R15-R16

TODO: add

### Model
TODO: add

### Design
TODO: add

## Leaderless Byzantine Fault Tolerance

> This type of consensus algorithm combines Snowball algorithm and Practical Byznatine Fault Tolerance (pBFT) algorithm. It tries to take advantage of decentralized aspect of Snowball and deterministic property of pBFT: eliminates the weakness of Snowballs's probabilistic property and pBFT reliance on a "leader".

Reviewer: ```suggestion
This type of consensus algorithm combines the Snowball and Practical Byznatine Fault Tolerance (pBFT) algorithms. It tries to take advantage of the decentralized aspect of Snowball and deterministic property of pBFT in order to eliminate the weakness of Snowballs's probabilistic property and pBFT's reliance on a "leader".
```

WDYT?

Author: Looks better to me, updated https://github.com/optakt/optakt.github.io/pull/13/commits/0a1cb902f3417c3bc20f9197df71d3d65a1b6421#diff-7b8cdfdde84c7352f51f3cc642d28c90f9d1283870319c81956812353289fb32R23-R26
